src/backend/cpu/kernel/ireduce.hpp

```diff
@@ -28,13 +30,17 @@ struct MinMaxOp
     MinMaxOp(T val, uint idx) :
         m_val(val), m_idx(idx)
     {
+        if (isNan(val)) {
+            m_val = Binary<T, op>::init();
+        }
     }
 
     void operator()(T val, uint idx)
     {
-        if (cabs(val) < cabs(m_val) ||
-            (cabs(val) == cabs(m_val) &&
-             idx > m_idx)) {
+        if (!isNan(val) &&
```

Comment:
I think if we reverse the order of the && operation we can avoid calling IS_NAN as much because of short-circuiting.

so instead of
```
if (!(IS_NAN(lhs[0])) &&
    ((sabs(lhs[0]) < sabs(rhs)) || (sabs(lhs[0]) == sabs(rhs)))) {
```
```
if ((sabs(lhs[0]) < sabs(rhs)) || (sabs(lhs[0]) == sabs(rhs)))
    && !(IS_NAN(lhs[0])))
```